components/Shipping/exceptions.rst

```diff
@@ -0,0 +1,40 @@
+Exceptions
+==========
+
+Calculator Registry Exceptions
+------------------------------
+
+.. _component_shipping_calculator_registry_existing-calculator-exception:
+
+ExistingCalculatorException
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This exception should is thrown when calculator with given name already exists.
+
+NonExistingCalculatorException
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This exception should is thrown when calculator with given name does not exist.
+
+UndefinedShippingMethodException
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This exception should is thrown when given shipment does not have a shipping method defined.
+
+Checker Registry Exceptions
+---------------------------
+
+ExistingRuleCheckerException
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This exception should is thrown when checker of given type already exists.
+
+NonExistingRuleCheckerException
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This exception should is thrown when checker of given type does not exist.
+
+.. note::
+    All of above exceptions extends the `PHP InvalidArgumentException`_.
+
+.. _PHP InvalidArgumentException: http://php.net/manual/en/class.invalidargumentexception.php
```

Comment:
It will be ok. Now there is caution in basic usage if a service throws exception . Should we describe it again in note (or in desc) below description of service?